.github/workflows/main.yml

```
@@ -0,0 +1,44 @@
+name: CI
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events but only for the main branch
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
```

```suggestion
  push:
    branches: [ main ]
    paths:                                                                      
      # run the tests when any python file is changed                           
      - '**.py'   
  pull_request:
    branches: [ main ]
    paths:                                                                      
      # run the tests when any python file is changed                           
      - '**.py'   
```
Adding this will prevent pytest from running for changes to a README, for example. (There might be more files/types we want to include though.)